RevenueCatUI/Templates/Components/Text/TextComponentViewModel.swift

```diff
@@ -16,50 +16,226 @@ import SwiftUI
+        }
+
+        // Apply medium on top of existing partial
+        if let medium = self.localizedOverrides?.conditions?.medium,
+           conditionTypesToApply.contains(.medium) {
+            combinedPartial = combine(combinedPartial, with: medium)
+        }
+
+        // Apply expanded on top of existing partial
+        if let expanded = self.localizedOverrides?.conditions?.expanded,
+           conditionTypesToApply.contains(.expanded) {
+            combinedPartial = combine(combinedPartial, with: expanded)
+        }
+
+        // Return the combined partial if it's not empty, otherwise return nil
+        return combinedPartial.partial.isEmpty ? nil : combinedPartial
     }
 
-    var horizontalAlignment: TextAlignment {
-        component.horizontalAlignment.textAlignment
+    private func combine(_ base: LocalizedTextPartial?, with other: LocalizedTextPartial?) -> LocalizedTextPartial {
+        let otherPartial = other?.partial
+        let basePartial = base?.partial
+
+        return LocalizedTextPartial(
+            text: other?.text ?? base?.text,
+            partial: PaywallComponent.PartialTextComponent(
+                visible: otherPartial?.visible ?? basePartial?.visible,
+                text: otherPartial?.text ?? basePartial?.text,
+                fontFamily: otherPartial?.fontFamily ?? basePartial?.fontFamily,
+                fontWeight: otherPartial?.fontWeight ?? basePartial?.fontWeight,
+                color: otherPartial?.color ?? basePartial?.color,
+                backgroundColor: otherPartial?.backgroundColor ?? basePartial?.backgroundColor,
+                padding: otherPartial?.padding ?? basePartial?.padding,
+                margin: otherPartial?.margin ?? basePartial?.margin,
+                textStyle: otherPartial?.textStyle ?? basePartial?.textStyle,
+                horizontalAlignment: otherPartial?.horizontalAlignment ?? basePartial?.horizontalAlignment
+            )
+        )
     }
 
-    var backgroundColor: Color {
-        component.backgroundColor?.toDyanmicColor() ?? Color.clear
+}
+
+extension LocalizedTextPartial {
+
+    static func create(
+        from partial: PaywallComponent.PartialTextComponent,
+        using localizedStrings: PaywallComponent.LocalizationDictionary
+    ) throws -> LocalizedTextPartial {
+        return LocalizedTextPartial(
+            text: try partial.text.flatMap { key in
+                try localizedStrings.string(key: key)
+            },
+            partial: partial
+        )
     }
 
-    var padding: EdgeInsets {
-        component.padding.edgeInsets
+}
+
+extension PaywallComponent.PartialTextComponent {
+
+    var isEmpty: Bool {
+        return visible == nil && text == nil && fontFamily == nil && fontWeight == nil && color == nil &&
+               backgroundColor == nil && padding == nil && margin == nil &&
+               textStyle == nil && horizontalAlignment == nil
```

Comment:
This might be hard to keep up to date if we ever add properties to `PartialTextComponent`. Maybe this should not be an extension but just a member of `PartialTextComponent`, so it's closer to the actual properties? For good measure, it might be good to also add a comment to `PartialTextComponent` instructing to update `isEmpty` if any properties are added.

Reply:
Ahhh yes! Will see if we can move this and might also be able to add some unit tests to validate it gets update 🤷‍♂️ 

I did this with new properties on full components and partials and it caught the new `shadow` property 😁 👇 

<img width="889" alt="Screenshot 2024-10-31 at 9 34 33 PM" src="https://github.com/user-attachments/assets/4f92d06b-6a41-4874-9efe-5d670ef91233">

Reply:
Oh that's very nice!